NOTES.md

```diff
@@ -0,0 +1,235 @@
+
+```
+{
+  "firstName": "John",
+  "lastName": "Smith",
+  "username": "JohnSmith",
+  "email": "John@Smith.com",
+  "password": "0123abc!@£",
+  "address": {
+	  "city": "Berlin"
+  }
+}
+```
+
+Will recieve the response =>
+
+```
+{
+  "avatar": "/statics/01.png",
+  "points": 0,
+  "booksToOffer": [],
+  "booksToRemember": [],
+  "booksInterestedIn": [],
+  "matches": [],
+  "_id": "60a26618f8d1360733111027",
+  "firstName": "John",
+  "lastName": "Smith",
+  "username": "JohnSmith",
+  "email": "John@Smith.com",
+  "address": {
+    "country": "Germany",
+    "city": "Berlin"
+  },
+  "createdAt": "2021-05-17T12:48:24.125Z",
+  "updatedAt": "2021-05-17T12:48:24.125Z"
+}
+``
+```
+
+### /:id
+
+**get** => response is an the user as an object
+**put** => If the field is made up =>
+
+```
+"This": "is not a field"
+```
+
+Will **not** return an error, just the user as an object not changed in any way. If you try and change a field that is a _mongoose id_, you will recieve an error. Fields that can be changed include =>
+
+```
+{
+  "avatar": "https://cdn.fakercloud.com/avatars/aislinnkelly_128.jpg",
+  "points": 2,
+  "firstName": "Mireya",
+  "lastName": "Armstrong",
+  "username": "Fred32",
+  "email": "Ayden9@hotmail.com",
+  "address": {
+    "country": "Germany",
+    "city": "Berlin"
+  },
+  "createdAt": "2020-05-18T00:00:00.000Z",
+  "updatedAt": "2021-05-18T05:17:48.493Z"
+}
+```
+
+**del** => if deleted successfully you will recieve a response of the user as an object, everything else will throw an error.
+
+### /login
+
+**post** => _to be updated_ => expects the following request =>
+
+```
+{
+	"email": "the users email"
+}
+```
+
+The response will be the user as an object.
+
+## Books
+
+All book calls start with /books
+
+### /
+
+**get** => to be cleared after testing, responseds with an array of all available book objects
+**post** => expects the following minimal request =>
+
+```
+{
+	"title": "xmen",
+	"subtitle": "The Big X",
+	"authors": ["J. K. Rowling"],
+	"publishedDate": 2002,
+	"pages": 10,
+	"shape": "good condition",
+	"category": ["fantasy"],
+	"selectedFiles": "file 03",
+	"owner": "a valid user mongoose id"
+}
+```
+
+_city will default to berlin_
+
+you will recieve the following response: an object with "newBook": the book as an object && "addBookToUser": the user who is the owner of that book =>
+
+```
+{
+  "newBook": {
+    "authors": [
+      "J. K. Rowling"
+    ],
+    "isbn": [],
+    "city": "Berlin",
+    "category": [
+      "fantasy"
+    ],
+    "selectedFiles": [
+      "file 03"
+    ],
+    "reserved": false,
+    "interestedUsers": [],
+    "_id": "60a351257062560d179d7d97",
+    "title": "xmen",
+    "subtitle": "The Big X",
+    "publishedDate": "2002",
+    "pages": 10,
+    "shape": "good condition",
+    "owner": "60a25ada6d711a9e70faa165",
+    "createdAt": "2021-05-18T05:31:17.831Z",
+    "updatedAt": "2021-05-18T05:31:17.831Z"
+  },
+  "addBookToUser": {
```

Comment:
this is a separate request?

Reply:
this happens at the same time the book is created because the owner need to have the book stored as the one it owns

Reply:
but do you want to keep it in the res.json? is the book with the new set owner enough for the frontend to work with?

Reply:
We can take it out, it is good to see that the book id has been added to the user but we don't need to send it back